code/modules/reagents/machinery/chem_dispenser.dm

```diff
@@ -316,6 +318,21 @@ USE THIS CHEMISTRY DISPENSER FOR MAPS SO THEY START AT 100 ENERGY
 			return
 
 	ui_interact(user)
+
+/obj/machinery/chem_dispenser/update_icon()
+
+	overlays.len = 0
+	var/image/overlay
+
+	if(istype(container, /obj/item/weapon/reagent_containers/glass/beaker/bluespace) || istype(container, /obj/item/weapon/reagent_containers/glass/beaker/noreact))
+		overlay = image('icons/obj/chemical.dmi', src, "dispenser_overlay2")
+	else
+		overlay = image('icons/obj/chemical.dmi', src, "dispenser_overlay1")
+
+	if(container)
```

Comment:
Sorry it's a bit of a nitpick but you should put everything except the `overlays.len = 0` under this check
No point in generating the image if it's not going to be used

Reply:
👍